src/ToolInfo.php

```diff
@@ -45,6 +45,7 @@ public function getComposerInstallationDetails(): array
         if (null === $this->composerInstallationDetails) {
             $composerInstalled = json_decode(file_get_contents($this->getComposerInstalledFile()), true, 512, JSON_THROW_ON_ERROR);
 
+            /** @var list<null|array{name: string, version: string, dist: array{reference?: string}}> $packages */
```

Comment:
can item of list really be `null` ?
if there would be `null`, wouldn't line 52 fail ?

Reply:
I've copied the type from the property - you are right, `null` should not be allowed here.